tst/testinstall/tilde.tst

```diff
@@ -46,4 +46,31 @@ Syntax error: ~ is not a valid name for an argument in stream:1
  ^
 gap> ({} -> ~);
 function(  ) ... end
+gap> list1 := [1,~];
+[ 1, ~ ]
+gap> list2 := [1,[1,[1,[1,0]]]];
+[ 1, [ 1, [ 1, [ 1, 0 ] ] ] ]
+
+# Check we increment / decrement stack correctly
```

Comment:
This is not clear as to what "stack" is meant. I assume this refers to the `RecursionDepth` variable in the kernel -- I think the comment should say so explicitly.

Also, I think this implicitly relies on the `RecursionTrapInterval` being "small" enough (right now it is set to 5000). That's fine, but I think I'd mention this, too. E.g.:

```
# Check that the RecursionDepth counter in the kernel is incremented and
# decremented correctly. If it isn't decremented correctly, then it will
# eventually exceed the `RecursionTrapInterval` (which normally is
# 5000), and the test will fail due to an unexpected error message.
```